macroid-core/src/main/scala/macroid/MultiEventTweaks.scala

```diff
@@ -0,0 +1,146 @@
+  }
+
+  def onMultipleFuncImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handlers: (c.Expr[String => Any])*) = {
+    import c.universe._
+
+    val tpe = weakTypeOf[W]
+    val (_, setter) = findSetter(c)(event, tpe)
+
+    val (listener, handlerOnMethodTupleSequence) = onBaseMultipleHandlers(c)(setter, handlers)
+
+    val overrides = handlerOnMethodTupleSequence map { case (handler, on, handlerType) =>
+      getOverride(c)(on, c.Expr(c.resetLocalAttrs(handler.tree.children.last)), handlerType)
+    }
+
+    c.Expr[Tweak[View]](q"_root_.macroid.Tweak[$tpe] { x ⇒ x.$setter(new $listener { ..$overrides })}")
+  }
+
+  def findSetter(c: MacroContext)(event: c.Expr[String], tp: c.Type) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(eventName: String))) = event
+    val setter = scala.util.Try {
+      val s = tp.member(newTermName(s"setOn${eventName.capitalize}Listener")).asMethod
+      assert(s != NoSymbol)
+      s
+    } getOrElse {
+      c.abort(c.enclosingPosition, s"Could not find method setOn${eventName.capitalize}Listener in $tp. " +
+          s"You may need to provide the type argument explicitly")
+    }
+    (eventName, setter)
+  }
+
+  def onBaseMultipleHandlers(c: MacroContext)(setter: c.universe.MethodSymbol, handlers: Seq[c.Expr[String => Any]]) = {
+    import c.universe._
+
+    val listener = setter.paramss(0)(0).typeSignature
+    val handlerOnMethodTupleSequence = handlers map { h =>
+      val (eventName, typ) = h.tree.children.head match {
+        case ValDef(_, name, tpt, _) => (name.toString, tpt)
+      }
+
+      val handlerType = typ.toString() match {
+        case s: String if s.endsWith("UnitHandler") => HUnit
+        case s: String if s.endsWith("FuncHandler") => HUnitFunc
+      }
+
+      val on = scala.util.Try {
+        val x = listener.member(newTermName(s"$eventName")).asMethod
+        assert(x != NoSymbol)
+        x
+      } getOrElse {
+        c.abort(c.enclosingPosition, s"Unsupported event listener class in $setter")
+      }
+
+      (h, on, handlerType)
+    }
+
+    val pendingMembersToOverride = listener.members filter (m => m.isMethod
+        && m.name.toString.startsWith("on")) toSet
+
+    val newSet = if (pendingMembersToOverride.nonEmpty) {
+      val specifiedMembers = handlerOnMethodTupleSequence map (t => t._2) toSet
+      val notImplemented = pendingMembersToOverride.asInstanceOf[Set[MethodSymbol]] diff specifiedMembers
+
+      val defaultHandlers = notImplemented map { m =>
+        val defaultHandler = c.Expr[Nothing](q"(${m.name.toTermName}: FuncHandler) => _root_.macroid.Ui")
+
+        (defaultHandler, m, HUnitFunc)
+      }
+
+      handlerOnMethodTupleSequence ++ defaultHandlers
+    } else handlerOnMethodTupleSequence
+
+    (listener, newSet)
+  }
+
+  def getOverride(c: MacroContext)(on: c.universe.MethodSymbol, f: c.Expr[Any], handlerType: HandlerType) = {
```

Comment:
This looks very much like `EventTweakMacros.getListener`. Could you adapt either of these two methods so that only one of them can be reused (same as the comments above)?